### DISCUSSION TITLE
Showing and Hiding a Collapse without a variable

### DISCUSSION THREAD

**Discussion by Maaaxiii** [Megabit/Blazorise#1848] [Q&A] [answered]
Hi there,

I was wondering if its possible to show and Hide an Collapse (in an Accordion) without having to define a dedicated bool. 

[The Documentation ](https://blazorise.com/docs/components/accordion/) only shows an example with dedicated bools. 

My Problem: I am generating a Collapse for each element in a List, so I am unable to define a dedicated bool. Is there any way I can still make my Collapse Visible and Invisible? 

Thanks for your Help

**Comment by stsrki** [comment, marked as answer]
No there is no way. Collapse is a pretty generic container/component and there is no other dedicated component that should act as close action for Collapse.

**Reply by Maaaxiii** [reply]
Thanks for the quick answer 🥇

**Reply by stsrki** [reply]
Maybe I should have checked before I posted an answer 😅

You have two ways of doing it actually.

1. Have `@ref` for collapse and then use `Toggle()`, eg. `collapse.Toggle()`
2. You can use the cascaded value from collapse(only if your other component is inside of Collapse).
```cs
  [CascadingParameter] public Collapse ParentCollapse { get; set; }
  
  void OnSomeClick()
  {
    ParentCollapse.Toggle()
  }
```
